### ISSUE TITLE
Load Image Not RUN

### ISSUE THREAD

**Issue by wjddyd66** [tensorflow/tensorflow#35355]
Yout Tutorial Link: https://www.tensorflow.org/tutorials/load_data/images?hl=ko

In Load using tf.data Section Code Error occur....

my Tensorflow Version: '2.0.0-beta1'

Code Here:
```python
def get_label(file_path):
    parts = tf.strings.split(file_path, os.path.sep)
    tf.print(parts, output_stream=sys.stderr)
    tf.print(parts[-2], output_stream=sys.stderr)
    
    return CLASS_NAMES == parts[-2]

def decode_img(img):
    # convert the compressed string to a 3D uint8 tensor
    img = tf.image.decode_jpeg(img, channels=3)
    # Use `convert_image_dtype` to convert to floats in the [0,1] range.
    img = tf.image.convert_image_dtype(img, tf.float32)
    # resize the image to the desired size.
    return tf.image.resize(img, [IMG_WIDTH, IMG_HEIGHT])

def process_path(file_path):
    label = get_label(file_path)
    # load the raw data from the file as a string
    img = tf.io.read_file(file_path)
    img = decode_img(img)
    return img, label

#AUTOTUNE = tf.data.experimental.AUTOTUNE

# Set `num_parallel_calls` so multiple images are loaded/processed in parallel.
labeled_ds = list_ds.map(process_path)

for image, label in labeled_ds.take(3):
    print("Image shape: ", image.numpy().shape)
    print("Label: ", label.numpy())
```

The result is
```code
["" "root" ".keras" ... "flower_photos" "dandelion" "17457028309_95514c8d02_n.jpg"]
dandelion
Image shape:  (224, 224, 3)
Label:  False
["" "root" ".keras" ... "flower_photos" "roses" "5537794501_a0767743fd_n.jpg"]
roses
Image shape:  (224, 224, 3)
Label:  False
["" "root" ".keras" ... "flower_photos" "dandelion" "4632235020_d00ce1e497.jpg"]
dandelion
Image shape:  (224, 224, 3)
Label:  False
```

**It's Version Mismatch or something wrong**

**Comment by ruchit2801** [comment]
I assume this is not an error. In my system and on colab, whole tutorial works great. The result you've uploaded is due to two print lines you've put in your `get_label` function. Essentially `parts` contains different parts of location of different images of your datasets. We are only interested in the labels right (apparent from the name of the function `get_label`) , which are names of parent directory of every file. So, for example every file have some name like `16149016979_23ef42b642_m.jpg`  and it is image of rose then it is stored in `roses` directory.

Now following line is path to image file stored in list,
`parts = tf.strings.split(file_path, os.path.sep)` 

 e.g. if `/home/.keras/datasets/flower-photos/roses/1.png` is location of image, then `parts` will be `["", "home", ".keras", "datasets", "flower-photos" ,"roses", "1.png"]`

Since we are interested in label, we choose `parts[-2]`. Since you are printing parts and parts[-2], you are getting an output out of this print statement every time `get_label` is called(So many lines). So, you just need to remove those two print statements from `get_label`, and code works great. 
@wjddyd66

**Comment by wjddyd66** [comment]
@ruchit2801 
Thanks for your answer.
But there is still a problem.....

First, <code>tf.print()</code> in <code>get_label</code> is just check it works....
I wrote the code again as your answer.

Code is 
```python
def get_label(file_path):
  # convert the path to a list of path components
  parts = tf.strings.split(file_path, os.path.sep)
  # The second to last is the class-directory
  return parts[-2] == CLASS_NAMES

def decode_img(img):
  # convert the compressed string to a 3D uint8 tensor
  img = tf.image.decode_jpeg(img, channels=3)
  # Use `convert_image_dtype` to convert to floats in the [0,1] range.
  img = tf.image.convert_image_dtype(img, tf.float32)
  # resize the image to the desired size.
  return tf.image.resize(img, [IMG_WIDTH, IMG_HEIGHT])

def process_path(file_path):
  label = get_label(file_path)
  # load the raw data from the file as a string
  img = tf.io.read_file(file_path)
  img = decode_img(img)
  return img, label

# Set `num_parallel_calls` so multiple images are loaded/processed in parallel.
labeled_ds = list_ds.map(process_path, num_parallel_calls=AUTOTUNE)

for image, label in labeled_ds.take(1):
  print("Image shape: ", image.numpy().shape)
  print("Label: ", label.numpy())
```

But the result is 
```code
Image shape:  (224, 224, 3)
Label:  False
```

In Upper Code I check  <code>parts[-2]</code> is Flower Directory
And Using <code>tf.print</code> Your answer is correct

So I Check CLASS_NAMES
```python
a = 'tulips'
print(a == CLASS_NAMES) # [False False False False  True]
```

What is Problem.........
Itroduction Output = [False False False False  True]
But My Code output Just [False]

![image](https://user-images.githubusercontent.com/37298751/71393074-19c70780-264e-11ea-892b-23d6ce526214.png)

**Comment by ruchit2801** [comment]
@wjddyd66, yes, then it should be a problem. I could not reproduce it, works for me. But, just to make sure, you have taken `CLASS_NAMES` as numpy array right? Version I am using is '2.0.0'.

**Comment by wjddyd66** [comment]
@ruchit2801  Hmm.....
Yes <code>CLASS_NAMES</code> as numpy array

Here is My Check Code
```python
print(tf.__version__)
CLASS_NAMES = np.array([item.name for item in data_dir.glob('*') if item.name != "LICENSE.txt"])
print(CLASS_NAMES)
print(type(CLASS_NAMES))
print(type(CLASS_NAMES[0]))
```
The Result is
```code
2.0.0-beta1
['dandelion' 'sunflowers' 'daisy' 'roses' 'tulips']
<class 'numpy.ndarray'>
<class 'numpy.str_'>
```
<br>

In Jupyter Notebook
![image](https://user-images.githubusercontent.com/37298751/71394503-374b9f80-2655-11ea-9384-b93487173871.png)

**Comment by ravikyram** [comment]
@wjddyd66 
I have tried on colab with TF version 2.0 beta1 and was able to reproduce the issue.However i am not seeing any issue with TF 2.0. Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/2fef781e3672efc80b454acd42e8ddb8/untitled497.ipynb) Thanks!

**Comment by wjddyd66** [comment]
@ravikyram, @ruchit2801  
Thank you, It resloved...
It's version probelm.....

I change Tnesorflow 2.0 beta1 -> Tensorflow 2.0 ... It works!

![image](https://user-images.githubusercontent.com/37298751/71398149-d62ac880-2662-11ea-90f9-c23caa5257c2.png)

**Comment by ravikyram** [comment]
I am closing the issue since its resolved. Thanks!.
